apps/chef_objects/src/chef_key.erl

```diff
@@ -147,29 +173,51 @@ list_query() ->
 create_query() ->
     insert_key_for_actor.
 
-parse_binary_json(_Bin, #chef_key{} = _ExistingObject) ->
-    % TODO according to rfc23, updates of individual fields are supported,
-    % so we'll validate each as optional and then verify that at least one is present.
-    error(unsupported);
-parse_binary_json(Bin, undefined) ->
-    EJ = chef_json:decode(Bin),
-
-    % validate public_key field
-    chef_object_base:validate_ejson(EJ, chef_object_base:public_key_spec(req)),
-
-    % validate name field
-    chef_object_base:validate_ejson(EJ, {[
-                                           {<<"name">>, {string_match, chef_regex:regex_for(key_name)}}
-                                         ]}),
-
-    % validate expiration_date and make is safe for sqerl
-    chef_object_base:validate_and_sanitize_date_field(EJ, <<"expiration_date">>).
+parse_binary_json(Bin, existing) ->
+  % In this case, any and all fields CAN be updated, but
+  % minimum one is required. Because there are only three possible fields,
+  % we can keep this simple:
+  EJ = chef_json:decode(Bin),
+  OneOf = [<<"expiration_date">>, <<"public_key">>, <<"name">>],
+  case [ej:get({Field}, EJ) || Field <- OneOf] of
+    [undefined,undefined,undefined] ->
```

Comment:
This might be annoying to fix in the future if somebody is like "here's two more fields we need to check"

maybe something like 
```erlang
case lists:all(fun(F) -> ej:get({F}, EJ) =:= undefined end, OneOf) of
    true ->
    false ->
end,
```